mayastor/tests/rebuild.rs

```diff
@@ -636,6 +636,191 @@ async fn rebuild_with_load() {
+
+    // Check the states of the nexus and children.
+    // Note: A failed rebuild will not change the state of the source child
+    // (even if it fails to read from it), but it will fault the destination
+    // child.
+    check_nexus_state(nexus_hdl, NexusState::NexusDegraded).await;
+    let rebuild_src = &get_share_uri(&ms2);
+    assert_eq!(
+        get_child_state(nexus_hdl, rebuild_src).await,
+        ChildState::ChildOnline as i32
+    );
+    assert_eq!(
+        get_child_state(nexus_hdl, rebuild_dst).await,
+        ChildState::ChildFaulted as i32
+    );
+}
+
+/// Test rebuild when restarting the destination container.
+#[tokio::test]
+async fn rebuild_restart_dst() {
+    let test = start_infrastructure("rebuild_restart_dst").await;
+    let (mut ms1, ms2, ms3) = setup_test(&test, 1).await;
+    let nexus_hdl = &mut ms1;
+    let rebuild_dst = &get_share_uri(&ms3);
+
+    // Check a rebuild is started for a newly added child.
+    add_child(nexus_hdl, rebuild_dst, true).await;
+    assert!(wait_for_rebuild_state(
+        nexus_hdl,
+        rebuild_dst,
+        "running",
+        Duration::from_secs(1),
+    )
+    .await
+    .unwrap());
+
+    // Restart the rebuild destination container and check the rebuild fails.
+    test.restart("ms3")
+        .await
+        .expect("Failed to restart rebuild destination");
+    assert_eq!(
+        wait_for_successful_rebuild(nexus_hdl, rebuild_dst).await,
+        false
+    );
+    assert_eq!(get_num_rebuilds(nexus_hdl).await, 0);
+
+    // Check the states of the nexus and children.
+    // Note: A failed rebuild will not change the state of the source child
+    // (even if it fails to read from it), but it will fault the destination
+    // child.
+    check_nexus_state(nexus_hdl, NexusState::NexusDegraded).await;
+    let rebuild_src = &get_share_uri(&ms2);
+    assert_eq!(
+        get_child_state(nexus_hdl, rebuild_src).await,
+        ChildState::ChildOnline as i32
+    );
+    assert_eq!(
+        get_child_state(nexus_hdl, rebuild_dst).await,
+        ChildState::ChildFaulted as i32
+    );
+}
+
+/// Test rebuild when disconnecting the source container from the network.
+#[tokio::test]
+async fn rebuild_src_disconnect() {
+    let test_name = "rebuild_src_disconnect";
+    let test = start_infrastructure(test_name).await;
+    let (mut ms1, ms2, ms3) = setup_test(&test, 1).await;
+    let nexus_hdl = &mut ms1;
+    let rebuild_dst = &get_share_uri(&ms3);
+
+    // Check a rebuild is started for a newly added child.
+    add_child(nexus_hdl, rebuild_dst, true).await;
+    assert!(wait_for_rebuild_state(
+        nexus_hdl,
+        rebuild_dst,
+        "running",
+        Duration::from_secs(1),
+    )
+    .await
+    .unwrap());
+
+    // Disconnect the rebuild source container from the network and check that
+    // the rebuild terminates. This requires a large timeout because it takes
+    // some time for the NVMf subsystem to report the error up.
```

Comment:
Could address this by reducing the timeouts in `nvme_bdev_opts`, that would reduce test runtime.

Reply:
Thanks for the suggestion. I gave it a go, changing the various times, but it didn't seem to make any noticeable difference in run time when I tried it locally. Perhaps I'm missing something. I think for now, I'll leave it as it is, but I'll raise a ticket for it. I think it would be worth revisiting.